src/qe_tools/extractors/__init__.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+from pathlib import Path
+
+from qe_tools.exceptions import PathIsNotAFile
+from qe_tools.parsers import CpInputFile, PwInputFile
+
+
+def extract(filepath: str, parser: str) -> dict:
+    """Extract QE input file as a dictionary.
+
+    Parameters
+    ----------
+    `filepath` : `str`
+        The path to the file.
+    `parser` : `str`
+        The QE parser type.
+
+    Returns
+    -------
+    `dict`
+        The parsed input as a dictionary.
+
+    Raises
+    ------
+    `FileNotFoundError`
+        If the file does not exist.
+    `PathIsNotAFile`
+        If the path does not point to a file.
+    `ValueError`
+        If the parser type is not supported.
+    """
+
+    path = Path(filepath)
+
+    if not path.exists():
+        raise FileNotFoundError(f"'{filepath}' does not exist")
+
+    if not path.is_file():
+        raise PathIsNotAFile(f"'{filepath}' is not a valid file.")
+
+    with open(path, encoding='utf-8') as input_file:
+        input_str = input_file.read()
+        parser = parser.upper()
+
+        if parser == 'PW':
+            return PwInputFile(input_str).as_dict()
+
+        if parser == 'CP':
+            return CpInputFile(input_str).as_dict()
+
+    raise ValueError(f"Unsupported parser argument: '{parser}'")
```

Comment:
Probably would be good to mention which values _are_ supported
```suggestion
    raise ValueError(f"Unsupported parser argument: '{parser}'. Please choose from {', '.join(['PW', 'CP'])}")
```